tests/tests_pytorch/trainer/test_config_validator.py

```diff
@@ -185,3 +185,34 @@ def custom_method(self, batch, *_, **__):
 
     with pytest.raises(MisconfigurationException, match=match_pattern):
         trainer.fit(model)
+
+
+@pytest.mark.parametrize(
+    "trainer_fn, dataloader_name",
+    [
+        ("fit", "train_dataloaders"),
+        ("fit", "val_dataloaders"),
+        ("validate", "dataloaders"),
+        ("test", "dataloaders"),
+        ("predict", "dataloaders"),
+    ],
+)
+def test_raise_exception_with_explicit_none_dataloader(trainer_fn, dataloader_name, tmpdir):
+    """Test that explicitly passing `Trainer.method(x_dataloader=None)` raises an error."""
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True)
+    model = BoringModel()
+
+    with pytest.raises(ValueError, match="You explicitly passed .*dataloaders=None.* but this is not supported"):
+        trainer_fn = getattr(trainer, trainer_fn)
```

Comment:
This should be outside the `raises` block
```suggestion
    trainer_fn = getattr(trainer, trainer_fn)
    with pytest.raises(ValueError, match="You explicitly passed .*dataloaders=None.* but this is not supported"):
```